pkg/operator/controller/status/controller.go

```diff
@@ -159,6 +185,10 @@ func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 		related = append(related, configv1.ObjectReference{
 			Resource: "namespaces",
 			Name:     state.IngressNamespace.Name,
+		}, configv1.ObjectReference{
+			Group:     iov1.GroupVersion.Group,
+			Resource:  "dnsrecords",
+			Namespace: state.IngressNamespace.Name,
```

Comment:
I didn't expect the addition of dnsRecords as a Gateway API related object.  Why is this needed?

Reply:
The gateway-service-dns controller puts dnsrecords in that namespace, so that it can put set owner reference on services in the same namespace.

Reply:
Why isn't this under the 

> if r.config.GatewayAPIEnabled {

condition?

Reply:
It doesn't need to be as the CRD exists anyway.  Furthermore, it makes sense to gather dnsrecords if they do somehow exist in the "openshift-ingress" namespace even without the featuregate enabled because the logic in the dns controller that reconciles dnsrecords in the "openshift-ingress" namespace isn't behind the featuregate.